In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

ruta_archivo = '../DataSets/european_leagues_player_stats/Big5-statsStats22-23.csv'

df = pd.read_csv(ruta_archivo, delimiter=',')

# Obtener información sobre el conjunto de datos
df.info()

# Calcular estadísticas descriptivas
df.describe()

# Ver las primeras filas del dataset
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2889 entries, 0 to 2888
Data columns (total 36 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Player      2889 non-null   object 
 1   Nation      2885 non-null   object 
 2   Pos         2889 non-null   object 
 3   Squad       2889 non-null   object 
 4   Comp        2889 non-null   object 
 5   Age         2889 non-null   int64  
 6   Born        2889 non-null   int64  
 7   MP          2889 non-null   int64  
 8   Starts      2889 non-null   int64  
 9   Min         2889 non-null   object 
 10  90s         2889 non-null   float64
 11  Gls         2889 non-null   int64  
 12  Ast         2889 non-null   int64  
 13  G+A         2889 non-null   int64  
 14  G-PK        2889 non-null   int64  
 15  PK          2889 non-null   int64  
 16  PKatt       2889 non-null   int64  
 17  CrdY        2889 non-null   int64  
 18  CrdR        2889 non-null   int64  
 19  xG          2872 non-null  

Player   Nation    Pos           Squad                Comp  Age  \
0  Brenden Aaronson   us USA  MF,FW    Leeds United  eng Premier League   21   
1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt       de Bundesliga   18   
2    James Abankwah   ie IRL     DF         Udinese          it Serie A   18   
3     George Abbott  eng ENG     MF       Tottenham  eng Premier League   16   
4  Yunis Abdelhamid   ma MAR     DF           Reims          fr Ligue 1   34   

   Born  MP  Starts    Min  ...  Gls.1  Ast.1  G+A.1  G-PK.1  G+A-PK  xG.1  \
0  2000  36      28  2,372  ...   0.04   0.11   0.15    0.04    0.15  0.15   
1  2003   7       0    173  ...   0.00   0.00   0.00    0.00    0.00  0.09   
2  2004   2       1     63  ...   0.00   0.00   0.00    0.00    0.00  0.00   
3  2005   1       0      1  ...   0.00   0.00   0.00    0.00    0.00  0.00   
4  1987  37      37  3,330  ...   0.03   0.05   0.08    0.03    0.08  0.06   

   xAG.1  xG+xAG  npxG.1  npxG+xAG.1  
0   0.16    0.31    0.15        0.31  
1   0.03    0.11    0.09        0.11  
2   0.00    0.00    0.00        0.00  
3   0.00    0.00    0.00        0.00  
4   0.03    0.09    0.06        0.09  

[5 rows x 36 columns]

#### Cambiar Valores
    
        David Ozoh:     Nation: 'eng ENG'


#### Borrar Filas
        
        Eliminar las filas con Nation = NaN
        
        Eliminar las filas con xG = NaN
        

#### Borrar Columnas

     26  Gls.1       2872 non-null   float64
     27  Ast.1       2872 non-null   float64
     28  G+A.1       2872 non-null   float64
     29  G-PK.1      2872 non-null   float64
     30  G+A-PK      2872 non-null   float64
     31  xG.1        2872 non-null   float64
     32  xAG.1       2872 non-null   float64
     33  xG+xAG      2872 non-null   float64
     34  npxG.1      2872 non-null   float64
     35  npxG+xAG.1  2872 non-null   float64

In [2]:
# Cambiar el valor del player David Ozoh y columna "Nation"
df.loc[2010, 'Nation'] = 'en ENG'

# Mostrar solo la fila 1875 del DataFrame
fila_2010 = df.iloc[2010:2013]
print(fila_2010)

                Player  Nation Pos           Squad                Comp  Age  \
2010        David Ozoh  en ENG  FW  Crystal Palace  eng Premier League   17   
2011    Levin Öztunalı  de GER  MF    Union Berlin       de Bundesliga   26   
2012  Fernando Pacheco  es ESP  GK         Almería          es La Liga   30   

      Born  MP  Starts Min  ...  Gls.1  Ast.1  G+A.1  G-PK.1  G+A-PK  xG.1  \
2010  2005   1       0   1  ...    0.0    0.0    0.0     0.0     0.0   0.0   
2011  1996   2       0   9  ...    0.0   10.0   10.0     0.0    10.0   0.0   
2012  1992   1       1  90  ...    0.0    0.0    0.0     0.0     0.0   0.0   

      xAG.1  xG+xAG  npxG.1  npxG+xAG.1  
2010   0.00    0.00     0.0        0.00  
2011   1.91    1.91     0.0        1.91  
2012   0.00    0.00     0.0        0.00  

[3 rows x 36 columns]


In [3]:
# Eliminar las filas con variable Nation = NaN
df = df.dropna(subset=['Nation'])

# Eliminar las filas con variable xG = NaN
df = df.dropna(subset=['xG'])

# Get the list of columns to be dropped
columns_to_drop = ['Gls.1', 'Ast.1', 'G+A.1', 'G-PK.1', 'G+A-PK', 'xG.1', 'xAG.1', 'xG+xAG', 'npxG.1', 'npxG+xAG.1']

# Drop the columns
df = df.drop(columns=columns_to_drop)

# Obtener información sobre el conjunto de datos
df.info()

# Calcular estadísticas descriptivas
df.describe()

# Ver las primeras filas del dataset
df.head()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2872 entries, 0 to 2885
Data columns (total 26 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Player    2872 non-null   object 
 1   Nation    2872 non-null   object 
 2   Pos       2872 non-null   object 
 3   Squad     2872 non-null   object 
 4   Comp      2872 non-null   object 
 5   Age       2872 non-null   int64  
 6   Born      2872 non-null   int64  
 7   MP        2872 non-null   int64  
 8   Starts    2872 non-null   int64  
 9   Min       2872 non-null   object 
 10  90s       2872 non-null   float64
 11  Gls       2872 non-null   int64  
 12  Ast       2872 non-null   int64  
 13  G+A       2872 non-null   int64  
 14  G-PK      2872 non-null   int64  
 15  PK        2872 non-null   int64  
 16  PKatt     2872 non-null   int64  
 17  CrdY      2872 non-null   int64  
 18  CrdR      2872 non-null   int64  
 19  xG        2872 non-null   float64
 20  npxG      2872 non-null   floa

Player   Nation    Pos           Squad                Comp  Age  \
0  Brenden Aaronson   us USA  MF,FW    Leeds United  eng Premier League   21   
1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt       de Bundesliga   18   
2    James Abankwah   ie IRL     DF         Udinese          it Serie A   18   
3     George Abbott  eng ENG     MF       Tottenham  eng Premier League   16   
4  Yunis Abdelhamid   ma MAR     DF           Reims          fr Ligue 1   34   

   Born  MP  Starts    Min  ...  PKatt  CrdY  CrdR   xG  npxG  xAG  npxG+xAG  \
0  2000  36      28  2,372  ...      0     2     0  3.9   3.9  4.2       8.1   
1  2003   7       0    173  ...      0     1     0  0.2   0.2  0.0       0.2   
2  2004   2       1     63  ...      0     0     0  0.0   0.0  0.0       0.0   
3  2005   1       0      1  ...      0     0     0  0.0   0.0  0.0       0.0   
4  1987  37      37  3,330  ...      0     6     0  2.4   2.4  1.0       3.4   

   PrgC   PrgP   PrgR  
0  43.0   86.0  151.0  
1   8.0    6.0   15.0  
2   0.0    0.0    0.0  
3   0.0    0.0    0.0  
4  40.0  215.0   10.0  

[5 rows x 26 columns]

## Guardar los datos en un archivo nuevo CSV

In [4]:
df.to_csv("../DataSets/european_leagues_player_stats/Big5-statsStats22-23_Corr.csv")

## Extraer los datos de nuevo archivo

In [5]:
ruta_archivo = '../DataSets/european_leagues_player_stats/Big5-statsStats22-23_Corr.csv'

df = pd.read_csv(ruta_archivo, delimiter=',')

# Obtener información sobre el conjunto de datos
df.info()

# Calcular estadísticas descriptivas
df.describe()

# Ver las primeras filas del dataset
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2872 entries, 0 to 2871
Data columns (total 27 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  2872 non-null   int64  
 1   Player      2872 non-null   object 
 2   Nation      2872 non-null   object 
 3   Pos         2872 non-null   object 
 4   Squad       2872 non-null   object 
 5   Comp        2872 non-null   object 
 6   Age         2872 non-null   int64  
 7   Born        2872 non-null   int64  
 8   MP          2872 non-null   int64  
 9   Starts      2872 non-null   int64  
 10  Min         2872 non-null   object 
 11  90s         2872 non-null   float64
 12  Gls         2872 non-null   int64  
 13  Ast         2872 non-null   int64  
 14  G+A         2872 non-null   int64  
 15  G-PK        2872 non-null   int64  
 16  PK          2872 non-null   int64  
 17  PKatt       2872 non-null   int64  
 18  CrdY        2872 non-null   int64  
 19  CrdR        2872 non-null  

Unnamed: 0            Player   Nation    Pos           Squad  \
0           0  Brenden Aaronson   us USA  MF,FW    Leeds United   
1           1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt   
2           2    James Abankwah   ie IRL     DF         Udinese   
3           3     George Abbott  eng ENG     MF       Tottenham   
4           4  Yunis Abdelhamid   ma MAR     DF           Reims   

                 Comp  Age  Born  MP  Starts  ... PKatt  CrdY  CrdR   xG  \
0  eng Premier League   21  2000  36      28  ...     0     2     0  3.9   
1       de Bundesliga   18  2003   7       0  ...     0     1     0  0.2   
2          it Serie A   18  2004   2       1  ...     0     0     0  0.0   
3  eng Premier League   16  2005   1       0  ...     0     0     0  0.0   
4          fr Ligue 1   34  1987  37      37  ...     0     6     0  2.4   

   npxG  xAG  npxG+xAG  PrgC   PrgP   PrgR  
0   3.9  4.2       8.1  43.0   86.0  151.0  
1   0.2  0.0       0.2   8.0    6.0   15.0  
2   0.0  0.0       0.0   0.0    0.0    0.0  
3   0.0  0.0       0.0   0.0    0.0    0.0  
4   2.4  1.0       3.4  40.0  215.0   10.0  

[5 rows x 27 columns]

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

ruta_archivo = '../DataSets/european_leagues_player_stats/Big5-shootingStats22-23.csv'

df = pd.read_csv(ruta_archivo, delimiter=',')

# Obtener información sobre el conjunto de datos
df.info()

# Calcular estadísticas descriptivas
df.describe()

# Ver las primeras filas del dataset
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2872 entries, 0 to 2871
Data columns (total 26 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  2872 non-null   int64  
 1   Player      2872 non-null   object 
 2   Nation      2872 non-null   object 
 3   Pos         2872 non-null   object 
 4   Squad       2872 non-null   object 
 5   Comp        2872 non-null   object 
 6   Age         2872 non-null   int64  
 7   Born        2872 non-null   int64  
 8   90s         2872 non-null   float64
 9   Gls         2872 non-null   int64  
 10  Sh          2872 non-null   int64  
 11  SoT         2872 non-null   int64  
 12  SoT%        2350 non-null   float64
 13  Sh/90       2872 non-null   float64
 14  SoT/90      2872 non-null   float64
 15  G/Sh        2350 non-null   float64
 16  G/SoT       1985 non-null   float64
 17  Dist        2350 non-null   float64
 18  FK          2872 non-null   float64
 19  PK          2872 non-null  

Unnamed: 0            Player   Nation    Pos           Squad  \
0           0  Brenden Aaronson   us USA  MF,FW    Leeds United   
1           1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt   
2           2    James Abankwah   ie IRL     DF         Udinese   
3           3     George Abbott  eng ENG     MF       Tottenham   
4           4  Yunis Abdelhamid   ma MAR     DF           Reims   

                 Comp  Age  Born   90s  Gls  ...  G/SoT  Dist   FK  PK  PKatt  \
0  eng Premier League   21  2000  26.4    1  ...   0.11  18.4  2.0   0      0   
1       de Bundesliga   18  2003   1.9    0  ...   0.00  18.7  0.0   0      0   
2          it Serie A   18  2004   0.7    0  ...    NaN   NaN  0.0   0      0   
3  eng Premier League   16  2005   0.0    0  ...    NaN   NaN  0.0   0      0   
4          fr Ligue 1   34  1987  37.0    1  ...   0.33  15.0  0.0   0      0   

    xG  npxG  npxG/Sh  G-xG  np:G-xG  
0  3.9   3.9     0.09  -2.9     -2.9  
1  0.2   0.2     0.04  -0.2     -0.2  
2  0.0   0.0      NaN   0.0      0.0  
3  0.0   0.0      NaN   0.0      0.0  
4  2.4   2.4     0.07  -1.4     -1.4  

[5 rows x 26 columns]

In [13]:
# Cambiar el valor del player David Ozoh y columna "Nation"
df.loc[2010, 'Nation'] = 'en ENG'

# Mostrar solo la fila 1875 del DataFrame
fila_2010 = df.iloc[2010:2013]
print(fila_2010)

# Eliminar las filas con variable Nation = NaN
df = df.dropna(subset=['Nation'])

# Eliminar las filas con variable xG = NaN
df = df.dropna(subset=['xG'])

# Obtener información sobre el conjunto de datos
df.info()

# Calcular estadísticas descriptivas
df.describe()

# Ver las primeras filas del dataset
df.head()

      Unnamed: 0                 Player  Nation    Pos           Squad  \
2010        2018    Helibelton Palacios  en ENG     DF           Elche   
2011        2019         Ante Palaversa  hr CRO     MF          Troyes   
2012        2020  Isaac Palazón Camacho  es ESP  FW,MF  Rayo Vallecano   

            Comp  Age  Born   90s  Gls  ...  G/SoT  Dist   FK  PK  PKatt   xG  \
2010  es La Liga   29  1993  21.3    0  ...   0.00  13.8  0.0   0      0  0.4   
2011  fr Ligue 1   22  2000   0.3    1  ...   1.00   5.2  0.0   0      0  0.5   
2012  es La Liga   27  1994  32.9    9  ...   0.36  24.6  3.0   1      3  7.7   

      npxG  npxG/Sh  G-xG  np:G-xG  
2010   0.4     0.09  -0.4     -0.4  
2011   0.5     0.51   0.5      0.5  
2012   5.3     0.06   1.3      2.7  

[3 rows x 26 columns]
<class 'pandas.core.frame.DataFrame'>
Int64Index: 2872 entries, 0 to 2871
Data columns (total 26 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  2

Unnamed: 0            Player   Nation    Pos           Squad  \
0           0  Brenden Aaronson   us USA  MF,FW    Leeds United   
1           1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt   
2           2    James Abankwah   ie IRL     DF         Udinese   
3           3     George Abbott  eng ENG     MF       Tottenham   
4           4  Yunis Abdelhamid   ma MAR     DF           Reims   

                 Comp  Age  Born   90s  Gls  ...  G/SoT  Dist   FK  PK  PKatt  \
0  eng Premier League   21  2000  26.4    1  ...   0.11  18.4  2.0   0      0   
1       de Bundesliga   18  2003   1.9    0  ...   0.00  18.7  0.0   0      0   
2          it Serie A   18  2004   0.7    0  ...    NaN   NaN  0.0   0      0   
3  eng Premier League   16  2005   0.0    0  ...    NaN   NaN  0.0   0      0   
4          fr Ligue 1   34  1987  37.0    1  ...   0.33  15.0  0.0   0      0   

    xG  npxG  npxG/Sh  G-xG  np:G-xG  
0  3.9   3.9     0.09  -2.9     -2.9  
1  0.2   0.2     0.04  -0.2     -0.2  
2  0.0   0.0      NaN   0.0      0.0  
3  0.0   0.0      NaN   0.0      0.0  
4  2.4   2.4     0.07  -1.4     -1.4  

[5 rows x 26 columns]

In [14]:
df.to_csv("../DataSets/european_leagues_player_stats/Big5-shootingStats22-23Corr.csv")

In [40]:
ruta_archivo_1 = '../DataSets/european_leagues_player_stats/Big5-statsStats22-23_Corr.csv'
ruta_archivo_2 = '../DataSets/european_leagues_player_stats/Big5-shootingStats22-23Corr.csv'

# Read the first CSV into a Pandas DataFrame
df1 = pd.read_csv(ruta_archivo_1)

# Read the second CSV into a Pandas DataFrame
df2 = pd.read_csv(ruta_archivo_2)

# Combine the two DataFrames
df_combined = df1.merge(df2, on='Player', how='outer')

# Calculate descriptive statistics
df_combined.describe()

# Keep only the specified columns
df_combined = df_combined[['Player', 'Nation_x', 'Pos_x', 'Squad_x', 'Comp_x', 'Age_x', 'Born_x', 'MP', 'Starts', 
                           'Min', '90s_x', 'Gls_x', 'Ast', 'G+A', 'G-PK', 'PK_x', 'PKatt_x', 'CrdY', 'CrdR', 'xG_x', 'npxG_x', 
                           'xAG', 'npxG+xAG', 'PrgC', 'PrgP', 'PrgR', 'Sh', 'SoT', 'SoT%', 'Sh/90', 'SoT/90', 'G/Sh', 
                           'G/SoT', 'Dist', 'FK', 'npxG/Sh', 'G-xG', 'np:G-xG']]



# Drop duplicate rows
df_combined = df_combined.drop_duplicates(subset='Player')

# Save the DataFrame to a CSV file
df_combined.to_csv('../DataSets/european_leagues_player_stats/combined_data.csv', index=False)

ruta_archivo = '../DataSets/european_leagues_player_stats/combined_data.csv'

# Read the combined data into a Pandas DataFrame
df = pd.read_csv(ruta_archivo, delimiter=',')

# Get information about the DataFrame
df.info()

# Calculate descriptive statistics
df.describe()

# See the first few rows of the DataFrame
df.head()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2701 entries, 0 to 2700
Data columns (total 38 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Player    2701 non-null   object 
 1   Nation_x  2701 non-null   object 
 2   Pos_x     2701 non-null   object 
 3   Squad_x   2701 non-null   object 
 4   Comp_x    2701 non-null   object 
 5   Age_x     2701 non-null   int64  
 6   Born_x    2701 non-null   int64  
 7   MP        2701 non-null   int64  
 8   Starts    2701 non-null   int64  
 9   Min       2701 non-null   object 
 10  90s_x     2701 non-null   float64
 11  Gls_x     2701 non-null   int64  
 12  Ast       2701 non-null   int64  
 13  G+A       2701 non-null   int64  
 14  G-PK      2701 non-null   int64  
 15  PK_x      2701 non-null   int64  
 16  PKatt_x   2701 non-null   int64  
 17  CrdY      2701 non-null   int64  
 18  CrdR      2701 non-null   int64  
 19  xG_x      2701 non-null   float64
 20  npxG_x    2701 non-null   floa

Player Nation_x  Pos_x         Squad_x              Comp_x  \
0  Brenden Aaronson   us USA  MF,FW    Leeds United  eng Premier League   
1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt       de Bundesliga   
2    James Abankwah   ie IRL     DF         Udinese          it Serie A   
3     George Abbott  eng ENG     MF       Tottenham  eng Premier League   
4  Yunis Abdelhamid   ma MAR     DF           Reims          fr Ligue 1   

   Age_x  Born_x  MP  Starts    Min  ...  SoT%  Sh/90  SoT/90  G/Sh  G/SoT  \
0     21    2000  36      28  2,372  ...  22.0   1.56    0.34  0.02   0.11   
1     18    2003   7       0    173  ...  25.0   2.08    0.52  0.00   0.00   
2     18    2004   2       1     63  ...   NaN   0.00    0.00   NaN    NaN   
3     16    2005   1       0      1  ...   NaN   0.00    0.00   NaN    NaN   
4     34    1987  37      37  3,330  ...   9.4   0.86    0.08  0.03   0.33   

   Dist   FK  npxG/Sh  G-xG  np:G-xG  
0  18.4  2.0     0.09  -2.9     -2.9  
1  18.7  0.0     0.04  -0.2     -0.2  
2   NaN  0.0      NaN   0.0      0.0  
3   NaN  0.0      NaN   0.0      0.0  
4  15.0  0.0     0.07  -1.4     -1.4  

[5 rows x 38 columns]

In [41]:
# Change the names of the variables
df.rename(columns={
    'Player_x': 'Player',
    'Nation_x': 'Nation',
    'Pos_x': 'Pos',
    'Squad_x': 'Squad',
    'Comp_x': 'Comp',
    'Age_x': 'Age',
    'Born_x': 'Born',
    '90s_x': '90s',
    'Gls_x': 'Gls',
    'Ast': 'Ast',
    'G+A': 'G+A',
    'G-PK': 'G-PK',
    'PK_x': 'PK',
    'PKatt_x': 'PKatt',
    'CrdY': 'CrdY',
    'CrdR': 'CrdR',
    'xG_x': 'xG',
    'npxG_x': 'npxG',
    'xAG': 'xAG',
    'npxG+xAG': 'npxG+xAG',
    'PrgC': 'PrgC',
    'PrgP': 'PrgP',
    'PrgR': 'PrgR',
    'Sh': 'Sh',
    'SoT': 'SoT',
    'SoT%': 'SoT%',
    'Sh/90': 'Sh/90',
    'SoT/90': 'SoT/90',
    'G/Sh': 'G/Sh',
    'G/SoT': 'G/SoT',
    'Dist': 'Dist',
    'FK': 'FK',
    'npxG/Sh': 'npxG/Sh',
    'G-xG': 'G-xG',
    'np:G-xG': 'np:G-xG'
}, inplace=True)

# Save the DataFrame to a CSV file
df.to_csv('../DataSets/european_leagues_player_stats/combined_data.csv', index=False)

ruta_archivo = '../DataSets/european_leagues_player_stats/combined_data.csv'

# Read the combined data into a Pandas DataFrame
df = pd.read_csv(ruta_archivo, delimiter=',')

# Get information about the DataFrame
df.info()

# Calculate descriptive statistics
df.describe()

# See the first few rows of the DataFrame
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2701 entries, 0 to 2700
Data columns (total 38 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Player    2701 non-null   object 
 1   Nation    2701 non-null   object 
 2   Pos       2701 non-null   object 
 3   Squad     2701 non-null   object 
 4   Comp      2701 non-null   object 
 5   Age       2701 non-null   int64  
 6   Born      2701 non-null   int64  
 7   MP        2701 non-null   int64  
 8   Starts    2701 non-null   int64  
 9   Min       2701 non-null   object 
 10  90s       2701 non-null   float64
 11  Gls       2701 non-null   int64  
 12  Ast       2701 non-null   int64  
 13  G+A       2701 non-null   int64  
 14  G-PK      2701 non-null   int64  
 15  PK        2701 non-null   int64  
 16  PKatt     2701 non-null   int64  
 17  CrdY      2701 non-null   int64  
 18  CrdR      2701 non-null   int64  
 19  xG        2701 non-null   float64
 20  npxG      2701 non-null   floa

Player   Nation    Pos           Squad                Comp  Age  \
0  Brenden Aaronson   us USA  MF,FW    Leeds United  eng Premier League   21   
1   Paxten Aaronson   us USA  MF,DF  Eint Frankfurt       de Bundesliga   18   
2    James Abankwah   ie IRL     DF         Udinese          it Serie A   18   
3     George Abbott  eng ENG     MF       Tottenham  eng Premier League   16   
4  Yunis Abdelhamid   ma MAR     DF           Reims          fr Ligue 1   34   

   Born  MP  Starts    Min  ...  SoT%  Sh/90  SoT/90  G/Sh  G/SoT  Dist   FK  \
0  2000  36      28  2,372  ...  22.0   1.56    0.34  0.02   0.11  18.4  2.0   
1  2003   7       0    173  ...  25.0   2.08    0.52  0.00   0.00  18.7  0.0   
2  2004   2       1     63  ...   NaN   0.00    0.00   NaN    NaN   NaN  0.0   
3  2005   1       0      1  ...   NaN   0.00    0.00   NaN    NaN   NaN  0.0   
4  1987  37      37  3,330  ...   9.4   0.86    0.08  0.03   0.33  15.0  0.0   

   npxG/Sh  G-xG  np:G-xG  
0     0.09  -2.9     -2.9  
1     0.04  -0.2     -0.2  
2      NaN   0.0      0.0  
3      NaN   0.0      0.0  
4     0.07  -1.4     -1.4  

[5 rows x 38 columns]

In [44]:
# Eliminar las filas con variable xG = NaN
df = df.dropna(subset=['G/Sh'])

# Get the list of columns to be dropped
columns_to_drop = ['G-xG', 'np:G-xG']

# Drop the columns
df = df.drop(columns=columns_to_drop)

# Save the DataFrame to a CSV file
df.to_csv('../DataSets/european_leagues_player_stats/combined_data.csv', index=False)

ruta_archivo = '../DataSets/european_leagues_player_stats/combined_data.csv'

# Read the combined data into a Pandas DataFrame
df = pd.read_csv(ruta_archivo, delimiter=',')

# Get information about the DataFrame
df.info()

# Calculate descriptive statistics
df.describe()

# See the first few rows of the DataFrame
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2206 entries, 0 to 2205
Data columns (total 36 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Player    2206 non-null   object 
 1   Nation    2206 non-null   object 
 2   Pos       2206 non-null   object 
 3   Squad     2206 non-null   object 
 4   Comp      2206 non-null   object 
 5   Age       2206 non-null   int64  
 6   Born      2206 non-null   int64  
 7   MP        2206 non-null   int64  
 8   Starts    2206 non-null   int64  
 9   Min       2206 non-null   object 
 10  90s       2206 non-null   float64
 11  Gls       2206 non-null   int64  
 12  Ast       2206 non-null   int64  
 13  G+A       2206 non-null   int64  
 14  G-PK      2206 non-null   int64  
 15  PK        2206 non-null   int64  
 16  PKatt     2206 non-null   int64  
 17  CrdY      2206 non-null   int64  
 18  CrdR      2206 non-null   int64  
 19  xG        2206 non-null   float64
 20  npxG      2206 non-null   floa

Player  Nation    Pos           Squad                Comp  Age  \
0   Brenden Aaronson  us USA  MF,FW    Leeds United  eng Premier League   21   
1    Paxten Aaronson  us USA  MF,DF  Eint Frankfurt       de Bundesliga   18   
2   Yunis Abdelhamid  ma MAR     DF           Reims          fr Ligue 1   34   
3      Himad Abdelli  dz ALG  MF,FW          Angers          fr Ligue 1   22   
4  Salis Abdul Samed  gh GHA     MF            Lens          fr Ligue 1   22   

   Born  MP  Starts    Min  ...  Sh  SoT  SoT%  Sh/90  SoT/90  G/Sh  G/SoT  \
0  2000  36      28  2,372  ...  41    9  22.0   1.56    0.34  0.02   0.11   
1  2003   7       0    173  ...   4    1  25.0   2.08    0.52  0.00   0.00   
2  1987  37      37  3,330  ...  32    3   9.4   0.86    0.08  0.03   0.33   
3  1999  30      24  2,137  ...  24    7  29.2   1.01    0.29  0.08   0.29   
4  2000  33      33  2,894  ...  13    4  30.8   0.40    0.12  0.08   0.25   

   Dist   FK  npxG/Sh  
0  18.4  2.0     0.09  
1  18.7  0.0     0.04  
2  15.0  0.0     0.07  
3  22.4  2.0     0.06  
4  19.1  0.0     0.07  

[5 rows x 36 columns]

### Variables Sinteticas

    Sh_per_90: Número de tiros por 90 minutos jugados. Esta variable mide la frecuencia con la que un jugador dispara al arco. Se puede calcular dividiendo la variable Sh por la variable 90s.
    
    SoT_per_90: Número de oportunidades de gol por 90 minutos jugados. Esta variable mide la frecuencia con la que un jugador crea oportunidades de gol. Se puede calcular dividiendo la variable SoT por la variable 90s.
    
    G_per_SoT: Número de goles por oportunidad de gol. Esta variable mide la efectividad de un jugador a la hora de convertir oportunidades de gol. Se puede calcular dividiendo la variable Gls por la variable SoT.


In [9]:
import pandas as pd
import numpy as np

ruta_archivo = '../DataSets/european_leagues_player_stats/combined_data.csv'

# Read the combined data into a Pandas DataFrame
df = pd.read_csv(ruta_archivo, delimiter=',')

df = df[['Player', 'Nation', 'Pos', 'Squad', 'Comp', 'Age', 'Starts', '90s', 'Gls', 'Sh', 'SoT', 'FK', 'Dist']]


In [10]:
# Create the `SoT_per_90` variable
df['Sh_per_90'] = df['Sh'] / df['90s']

# Create the `SoT_per_90` variable
df['SoT_per_90'] = df['SoT'] / df['90s']

# Create the `G_per_SoT` variable
df['G_per_SoT'] = df['Gls'] / df['SoT']

# Eliminar valores NaN
df['SoT_per_90'].replace(np.NAN, 0, inplace=True)
df['G_per_SoT'].replace(np.NAN, 0, inplace=True)
df['Sh_per_90'].replace(np.NAN, 0, inplace=True)

# Eliminar valores infinitos
df['SoT_per_90'].replace(np.inf, 0, inplace=True)
df['G_per_SoT'].replace(np.inf, 0, inplace=True)
df['Sh_per_90'].replace(np.inf, 0, inplace=True)

# Eliminar valores demasiado grandes
df['SoT_per_90'].replace(np.nan, 0, inplace=True)
df['G_per_SoT'].replace(np.nan, 0, inplace=True)
df['Sh_per_90'].replace(np.nan, 0, inplace=True)

# Save the DataFrame to a CSV file
df.to_csv('../DataSets/european_leagues_player_stats/sintetic_data.csv', index=False)

ruta_archivo = '../DataSets/european_leagues_player_stats/sintetic_data.csv'

# Read the combined data into a Pandas DataFrame
df = pd.read_csv(ruta_archivo, delimiter=',')

# Get information about the DataFrame
df.info()

# Calculate descriptive statistics
df.describe()

# See the first few rows of the DataFrame
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2206 entries, 0 to 2205
Data columns (total 16 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Player      2206 non-null   object 
 1   Nation      2206 non-null   object 
 2   Pos         2206 non-null   object 
 3   Squad       2206 non-null   object 
 4   Comp        2206 non-null   object 
 5   Age         2206 non-null   int64  
 6   Starts      2206 non-null   int64  
 7   90s         2206 non-null   float64
 8   Gls         2206 non-null   int64  
 9   Sh          2206 non-null   int64  
 10  SoT         2206 non-null   int64  
 11  FK          2206 non-null   float64
 12  Dist        2206 non-null   float64
 13  Sh_per_90   2206 non-null   float64
 14  SoT_per_90  2206 non-null   float64
 15  G_per_SoT   2206 non-null   float64
dtypes: float64(6), int64(5), object(5)
memory usage: 275.9+ KB


Player  Nation    Pos           Squad                Comp  Age  \
0   Brenden Aaronson  us USA  MF,FW    Leeds United  eng Premier League   21   
1    Paxten Aaronson  us USA  MF,DF  Eint Frankfurt       de Bundesliga   18   
2   Yunis Abdelhamid  ma MAR     DF           Reims          fr Ligue 1   34   
3      Himad Abdelli  dz ALG  MF,FW          Angers          fr Ligue 1   22   
4  Salis Abdul Samed  gh GHA     MF            Lens          fr Ligue 1   22   

   Starts   90s  Gls  Sh  SoT   FK  Dist  Sh_per_90  SoT_per_90  G_per_SoT  
0      28  26.4    1  41    9  2.0  18.4   1.553030    0.340909   0.111111  
1       0   1.9    0   4    1  0.0  18.7   2.105263    0.526316   0.000000  
2      37  37.0    1  32    3  0.0  15.0   0.864865    0.081081   0.333333  
3      24  23.7    2  24    7  2.0  22.4   1.012658    0.295359   0.285714  
4      33  32.2    1  13    4  0.0  19.1   0.403727    0.124224   0.250000